binarytree/__init__.py

```diff
@@ -314,6 +315,38 @@ def _get_tree_properties(root):
     }
 
 
+def _get_parent_node(root, node):
+    """Search from the binary tree and return the parent node for require node.
+
+    :param root: Root node of the binary tree.
+    :rtype: binarytree.Node
+    :param node: Require node you want to get its parent node.
+    :rtype: binarytree.Node
+    :return: The parent node of require node.
+    :rtype: binarytree.Node
+    """
+    root_height = root.height
+    node_height = node.height
+    if root_height == node_height:
+        return None
+    node_level_list = root.levels[root_height - node_height]
+    parent_level_list = root.levels[root_height - node_height - 1]
+    node_index = 0
+    for index in range(len(node_level_list)):
+        if node_level_list[index].value == node.value:
+            node_index = index
+            break
+    start_index = node_index // 2
+    for pnode in parent_level_list[start_index:]:
+        try:
+            if pnode.left.value == node.value or\
```

Comment:
What if there are multiple nodes with same values? Best way to check for object equality is using Python's `is` keyword.

Reply:
That's ok to use `is` keyword and I will modify it. In my cases all nodes are with different values so I didn't consider about this.